lib/puppet/provider/cisco_interface/cisco.rb

```diff
@@ -177,16 +143,31 @@
 
   def initialize(value={})
     super(value)
-    @nu = Cisco::Interface.interfaces[@property_hash[:name]]
+    if value.is_a?(Hash)
+      # value is_a hash when initialized from properties_get()
+      all_intf = value[:all_intf]
+      single_intf = value[:interface]
+    else
+      # note: @nu instantiation will be skipped
```

Comment:
Why is @nu skipped in this case?

Reply:
Clarify comment here.  What is really happening?

Reply:
I'll replace the comment with this:
```
      # @property_hash[:name] is nil in this codepath; because it's nil
      # it will cause @nu to become nil, thus @nu instantiation is just
      # skipped altogether.
```